lib/Rex/Commands/File.pm

```diff
@@ -1016,7 +1016,7 @@ sub delete_lines_according_to {
 
 }
 
-=head2 append_if_no_such_line($file, $new_line, @regexp)
+=head2 append_if_no_such_line($file, $new_line[, @regexp])
```

Comment:
As a general rule, I aim for going towards unified formatting style (including docs) in the largest scope possible (codebase, or at least within the files, or functions).

The formatting of optional and/or trailing parameters seems to be inconsistent throughout the codebase, but there's an example for a similar an optional trailing parameter in Rex::Commands::File: [`extract($file [, %options])`](https://metacpan.org/pod/Rex::Commands::File#extract($file-%5B,-%25options%5D)).

To have consistent formatting for the same concepts within the same file, what do you think about adding one more space and write it as `($file, $new_line [, @regexp])`?

Reply:
Apparently, for `append_if_no_such_line` there is an old format with positional parameters (like here), and a new (preferred) format with named parameters (like `regexp => ...`).

The old format supports optionally passing a single regular expression or multiple ones as an arrayref. It also supports optional `on_change` and `on_no_change` hooks. Some of that is also apparently missing from the detailed docs.

It is not clear how it would be best to address all those variations in a unified format, or whether thinking about these would be in the scope of this story or not.

Removing one point of confusion is certainly a step forward, so let's push this towards merging.

I just wanted to mention the above "accidental complexity" because I think it's a good example of why discussing a planned change first is almost always a good idea in this repo's context. "Open an issue before investing work in a PR" is the translation of that idea to GitHub workflow ;)